BetterHolidays/markets/holidays.py

```
@@ -42,82 +85,189 @@ def __init__(self, days:'list[int]', change:'dict[int, int]', start: dt.date = N
+class ChineseNewYearsDay(CommonHoliday):
+    name = "Chinese New Year's Day"
+
+    def get_date(self, year: "int") -> 'list[dt.date]':
+        page = md.HTML.from_url(f"https://www.timeanddate.com/calendar/?year={year}&country=41")
+        cny = [get_date_from_timeanddate(td, year) for td in page.inner_html.get_by_attr("title", "Chinese New Year")]
+        return flatten_list([cny[0] - dt.timedelta(last_day(cny[0].weekday(), SUNDAY)), cny, cny[-1]+dt.timedelta(days=next_weekend(cny[-1].weekday()))])
+
+class ChineseNationalDay(CommonHoliday):
+    name = "Chinese National Day"
+
+    def get_date(self, year: int) -> dt.date | list[date]:
```

Ruff (0.8.2)

Undefined name `date`

(F821)